tooling/cli/lib/deploy.js

```diff
@@ -57,68 +57,86 @@ const JOB_QUERY = /* GraphQL */ `
   }
 `;
 
+class DeploymentError extends Error {
+  #payload;
```

Comment:
using a custom error to hold additional error payload here. the `payload` property contains the batch job id and TI instance object. to avoid printing the payload info at Ln 127 below, declare it as a private class member here.